printer/printer_test.go

```diff
@@ -23,31 +23,55 @@ func TestPrinter(t *testing.T) {
 		if strings.HasSuffix(path, ".in") {
 			inputFile := path
 			expectFile := strings.TrimSuffix(path, ".in") + ".expect"
-
-			inputBB, err := os.ReadFile(inputFile)
-			require.NoError(t, err)
-			expectBB, err := os.ReadFile(expectFile)
-			require.NoError(t, err)
+			expectErrorFile := strings.TrimSuffix(path, ".in") + ".error"
 
 			caseName := filepath.Base(path)
 			caseName = strings.TrimSuffix(caseName, ".in")
 
 			t.Run(caseName, func(t *testing.T) {
-				testPrinter(t, inputBB, expectBB)
+				testPrinter(t, inputFile, expectFile, expectErrorFile)
 			})
 		}
 
 		return nil
 	})
 }
 
-func testPrinter(t *testing.T, input, expect []byte) {
-	f, err := parser.ParseFile(t.Name()+".rvr", input)
+func testPrinter(t *testing.T, inputFile string, expectFile string, expectErrorFile string) {
+	inputBB, err := os.ReadFile(inputFile)
+	require.NoError(t, err)
+
+	f, err := parser.ParseFile(t.Name()+".rvr", inputBB)
+	if expectedError := getExpectedErrorMessage(t, expectErrorFile); expectedError != "" {
+		require.Error(t, err)
+		require.Contains(t, err.Error(), expectedError)
```

Comment:
Doing a contains for now so we can test the interesting part of the error. This could become an equal at the expense of the tests being a little more brittle.